### DISCUSSION TITLE
Is it possible to support UDP GSO in libuv ?

### DISCUSSION THREAD

**Discussion by MrJia1997** [libuv/libuv#3348] [Ideas]
Generic segmentation offload, or GSO, has been proved to be an efficient way to increase UDP throughput. It was introduced to Linux kernel since version 4.18. 

Because we're heavily using libuv in our infrastructure, every bit of performance gain will be highly valued. Is it possible to add UDP GSO support in libuv?

**Comment by bnoordhuis** [comment]
In a word: no. Libuv is a platform abstraction layer but GSO is Linux-specific.<sup>1</sup>

What you can do is create a UDP socket manually, call `setsockopt(fd, SOL_UDP, UDP_SEGMENT, &size, sizeof(size)))`, then hand off the socket to `uv_udp_open()`. You also need to turn on GSO on the NIC with ethtool(8) if you haven't already.

<sup>1</sup> Windows also supports GSO but not on a per-socket level, as far as I'm aware, it's a NIC setting.
